bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java

```diff
@@ -417,30 +399,24 @@ private Object getRadioThermostatData(String resource) {
 
         try {
             // Run the HTTP request and get the JSON response from the thermostat
-            URL url = new URL(urlStr);
-            URLConnection connection = url.openConnection();
-
-            try {
-                String response = IOUtils.toString(connection.getInputStream());
-                logger.debug("thermostatResponse = {}", response);
-
-                // Map the JSON response to the correct object
-                if (DEFAULT_RESOURCE.equals(resource)) {
-                    result = gson.fromJson(response, RadioThermostatJsonResponse.class);
-                } else if (HUMIDITY_RESOURCE.equals(resource)) {
-                    result = gson.fromJson(response, RadioThermostatJsonHumidity.class);
-                } else if (RUNTIME_RESOURCE.equals(resource)) {
-                    result = gson.fromJson(response, RadioThermostatJsonRuntime.class);
-                } else if (MODEL_RESOURCE.equals(resource)) {
-                    result = gson.fromJson(response, RadioThermostatJsonModel.class);
-                } else if (NAME_RESOURCE.equals(resource)) {
-                    result = gson.fromJson(response, RadioThermostatJsonName.class);
-                }
-                
-            } finally {
-                IOUtils.closeQuietly(connection.getInputStream());
+            ContentResponse contentResponse = httpClient.newRequest(urlStr).method(GET).timeout(20, TimeUnit.SECONDS).send();
+
+            String response = contentResponse.getContentAsString();
+            logger.debug("thermostatResponse = {}", response);
+
+            // Map the JSON response to the correct object
+            if (DEFAULT_RESOURCE.equals(resource)) {
+                result = gson.fromJson(response, RadioThermostatJsonResponse.class);
+            } else if (HUMIDITY_RESOURCE.equals(resource)) {
+                result = gson.fromJson(response, RadioThermostatJsonHumidity.class);
+            } else if (RUNTIME_RESOURCE.equals(resource)) {
+                result = gson.fromJson(response, RadioThermostatJsonRuntime.class);
+            } else if (MODEL_RESOURCE.equals(resource)) {
+                result = gson.fromJson(response, RadioThermostatJsonModel.class);
+            } else if (NAME_RESOURCE.equals(resource)) {
+                result = gson.fromJson(response, RadioThermostatJsonName.class);
             }
-
+                
```

Comment:
There are some formatting errors. You can view them with `mvn install -Dspotless.check.skip=false` and fix them by running `mvn spotless:apply`.

Reply:
Done!